quickjs.h

```diff
@@ -562,30 +567,30 @@ JS_EXTERN JSValue JS_GetException(JSContext *ctx);
 JS_EXTERN JS_BOOL JS_IsError(JSContext *ctx, JSValue val);
 JS_EXTERN void JS_ResetUncatchableError(JSContext *ctx);
 JS_EXTERN JSValue JS_NewError(JSContext *ctx);
-JS_EXTERN JSValue __js_printf_like(2, 3) JS_ThrowSyntaxError(JSContext *ctx, const char *fmt, ...);
-JS_EXTERN JSValue __js_printf_like(2, 3) JS_ThrowTypeError(JSContext *ctx, const char *fmt, ...);
-JS_EXTERN JSValue __js_printf_like(2, 3) JS_ThrowReferenceError(JSContext *ctx, const char *fmt, ...);
-JS_EXTERN JSValue __js_printf_like(2, 3) JS_ThrowRangeError(JSContext *ctx, const char *fmt, ...);
-JS_EXTERN JSValue __js_printf_like(2, 3) JS_ThrowInternalError(JSContext *ctx, const char *fmt, ...);
+JS_EXTERN JSValue js_printf_like(2, 3) JS_ThrowSyntaxError(JSContext *ctx, const char *fmt, ...);
+JS_EXTERN JSValue js_printf_like(2, 3) JS_ThrowTypeError(JSContext *ctx, const char *fmt, ...);
+JS_EXTERN JSValue js_printf_like(2, 3) JS_ThrowReferenceError(JSContext *ctx, const char *fmt, ...);
+JS_EXTERN JSValue js_printf_like(2, 3) JS_ThrowRangeError(JSContext *ctx, const char *fmt, ...);
+JS_EXTERN JSValue js_printf_like(2, 3) JS_ThrowInternalError(JSContext *ctx, const char *fmt, ...);
 JS_EXTERN JSValue JS_ThrowOutOfMemory(JSContext *ctx);
 
-JS_EXTERN void __JS_FreeValue(JSContext *ctx, JSValue v);
+JS_EXTERN void JS_FreeValue__(JSContext *ctx, JSValue v);
 static inline void JS_FreeValue(JSContext *ctx, JSValue v)
 {
     if (JS_VALUE_HAS_REF_COUNT(v)) {
         JSRefCountHeader *p = (JSRefCountHeader *)JS_VALUE_GET_PTR(v);
         if (--p->ref_count <= 0) {
-            __JS_FreeValue(ctx, v);
+            JS_FreeValue__(ctx, v);
         }
     }
 }
-JS_EXTERN void __JS_FreeValueRT(JSRuntime *rt, JSValue v);
+JS_EXTERN void JS_FreeValueRT__(JSRuntime *rt, JSValue v);
```

Comment:
I don't understand the name change, if we are not moving it to the C file.

Reply:
Identifiers starting with `__` are reserved, hence the name change.

`JS_FreeValueRT` and `JS_FreeValue` are inline functions that test whether the value is refcounted.  I am not sure we should make them simple entry points and only implement them inline for the implementation.

Reply:
I'd leave that change for a subsequent PR, since it's unrelated to the JS_NewNumber changes.

Reply:
> I'd leave that change for a subsequent PR, since it's unrelated to the JS_NewNumber changes.

Yes, I should make separate pull requests.  This work in progress is only half baked.